File: ABC258/D_Trophy/258D220712.cpp
#include <iostream>
#include <string>
#include <vector>
 
#define ll long long
 
using namespace std;
 
 
int main(void) {
    int n, x;
    cin >> n >> x;
    vector<int> a(n), b(n);
    ll ans = numeric_limits<ll>::max();
    ll sum = 0;
    
    for(int i = 0 ; i < n ; i++){
        cin >> a[i] >> b[i];
    }
    
    for(int i = 0 ; i < n ; i++){
        x--;
        if(x == 0){
            break;
        }
        sum += (a[i] + b[i]);
        ans = min(ans, sum + (ll)b[i] * x);
    }
    
    cout << ans << endl;
    
    
    return 0;
}
 

File: ABC258/D_Trophy/258D220712.py
import math 
 
def main():
    n, x = map(int, input().split())
    sm = 0
    jeux = [list(map(int, input().split())) for _ in range(n)]
    ans = math.inf 
    
    for i in range(n):
        game = jeux[i]
        x -= 1
    
        if x == 0:
            break 
        
        sm += sum(game)
        ans = min(ans, sm + game[1] * x)
        
            
  
    print(ans)
    return 
            
        
 
 
if __name__ == "__main__":
    main()
